applications/MeshingApplication/custom_utilities/tetrahedra10_mesh_converter_utility.h

```diff
@@ -0,0 +1,356 @@
+
+        Triangle3D6<Node<3> > geom(
+                this_model_part.Nodes()(i0),
+                this_model_part.Nodes()(i1),
+                this_model_part.Nodes()(i2),
+                this_model_part.Nodes()(i3),
+                this_model_part.Nodes()(i4),
+                this_model_part.Nodes()(i5)
+                );
+        return geom;
+    }
+
+        /**
+    * It erases the old Tetrahedra3D4 elements and it creates the new Tetrahedra3D10 ones
+    * @param Coord: The compressed matrix containing at (i,j) the id of the node created between nodes i,j
+    * @param New_Elements: The new elements created
+    * @param interpolate_internal_variables: A boolean that defines if it is necessary to interpolate the internal variables
+    * @return this_model_part: The model part of the model (it is the input too)
+    */
+
+    void EraseOldElementAndCreateNewElement(
+            ModelPart& this_model_part,
+            const compressed_matrix<int>& Coord,
+            PointerVector< Element >& NewElements,
+            bool interpolate_internal_variables
+    ) override
+    {
+	ElementsArrayType& rElements = this_model_part.Elements();
+	ElementsArrayType::iterator it_begin = rElements.ptr_begin();
+	ElementsArrayType::iterator it_end = rElements.ptr_end();
+
+	const ProcessInfo& rCurrentProcessInfo = this_model_part.GetProcessInfo();
+	int edge_ids[6];
+	std::vector<int> node_ids;
+
+	for (ElementsArrayType::iterator it = it_begin; it != it_end; ++it)
+	{
+        GlobalPointersVector< Element >& rChildElements = it->GetValue(NEIGHBOUR_ELEMENTS);
+        rChildElements.resize(0);
+
+        CalculateEdges(it->GetGeometry(), Coord, edge_ids, node_ids);
+
+        // Generate the new Tetrahedra3D10 element
+        Tetrahedra3D10<Node<3>> geom = GenerateTetrahedra(this_model_part, node_ids);
+        const Element& rElem = KratosComponents<Element>::Get("Element3D10N");
+        Element::Pointer p_element;
+        p_element = rElem.Create(it->Id(), geom, it->pGetProperties());
+        p_element->Initialize(rCurrentProcessInfo);
+        p_element->InitializeSolutionStep(rCurrentProcessInfo);
+        p_element->FinalizeSolutionStep(rCurrentProcessInfo);
+
+        // Setting the internal variables in the "child" elem (the element replacing the old one)
+        if (interpolate_internal_variables == true)
+        {
+            //This method only copies the current information to the new element
+            InterpolateInteralVariables(0, *it.base(), p_element, rCurrentProcessInfo);
+        }
+
+        // Transfer elemental variables to new element
+        p_element->Data() = it->Data();
+        p_element->GetValue(SPLIT_ELEMENT) = false;
+        NewElements.push_back(p_element);
+
+        rChildElements.push_back( Element::WeakPointer(p_element) );
+    }
+
+    // Now replace the elements in SubModelParts
+    if ( NewElements.size() > 0 ) ReplaceElementsInSubModelPart(this_model_part);
```

Comment:
Here is ok because you have a vector, but in general the `.empty()` is more efficient, especially for lists and trees.

Reply:
Styling, one line if and loops are not encouraged.